motoman_driver/MotoPlus/CtrlGroup.c

```diff
@@ -277,17 +309,24 @@ void Ros_CtrlGroup_ConvertToRosPos(CtrlGroup* ctrlGroup, long pulsePos[MAX_PULSE
 		for(i=0; i<ctrlGroup->numAxes; i++)
 		{
 			if(i<2)
-				radPos[i] = pulsePos[i] / ctrlGroup->pulseToRad.PtoR[i];
+				rosPos[i] = pulsePos[i] / ctrlGroup->pulseToRad.PtoR[i];
 			else if(i==2)
-				radPos[2] = pulsePos[6] / ctrlGroup->pulseToRad.PtoR[6];
+				rosPos[2] = pulsePos[6] / ctrlGroup->pulseToRad.PtoR[6];
 			else
-				radPos[i] = pulsePos[i-1] / ctrlGroup->pulseToRad.PtoR[i-1];		
+				rosPos[i] = pulsePos[i-1] / ctrlGroup->pulseToRad.PtoR[i-1];
 		}
 	}
 	else
 	{
-		for(i=0; i<ctrlGroup->numAxes; i++)
-			radPos[i] = pulsePos[i] / ctrlGroup->pulseToRad.PtoR[i];
+		for (i = 0; i < ctrlGroup->numAxes; i++)
+		{
+			if (ctrlGroup->axisType.type[i] == ROTATATION_AXIS)
+				conversion = ctrlGroup->pulseToRad.PtoR[i];
+			else if (ctrlGroup->axisType.type[i] == LINEAR_AXIS)
+				conversion = ctrlGroup->pulseToMeter.PtoM[i];
+
+			rosPos[i] = pulsePos[i] / conversion;

```

Comment:
If the type is set incorrectly, then this could be a divide by zero.  I would recommend setting conversion to `1.0` just as a defensive coding measure.